src/net/tcp/stream.rs

```diff
@@ -123,6 +124,40 @@ impl TcpStream {
         self.inner.nodelay()
     }
 
+    /// Sets the value for the `SO_LINGER` option on this socket.
+    ///
+    /// When set, the closing/shutting down of the socket will not return
+    /// until all queued messages for the socket have been successfully
+    /// sent or the linger timeout has been reached.
+    pub fn set_linger(&self, dur: Option<Duration>) -> io::Result<()> {
```

Comment:
I think we should stick to the same API as std lib.

Reply:
std lib doesn't have `set_linger` option anymore. How else can we expose this feature?

I think the plan for tokio's TcpStream was to also stick to the same api as std lib but @carllerche says they've abandoned that plan.

Reply:
So for tokio::net `TcpStream` @carllerche mentioned to me the plan was to follow std lib's api but it was abandoned.
I assumed we want mio's `TcpStream` to be similar to tokio's.

If we only add `From` traits here then we'd still have to have deal with `ManuallyDrop` at tokio `TcpStream` layer.

Reply:
@Thomasdezeeuw what is the std lib API? I don't see one? Could you clarify what API you mean?

Reply:
@carllerche that is my point. Not having `set_linger` on `TcpStream`.